src/installer/installer.go

```diff
@@ -588,3 +600,20 @@ func (i *installer) createSingleNodeMasterIgnition() (string, error) {
 	}
 	return singleNodeMasterIgnitionPath, nil
 }
+
+func (i *installer) checkLocalhostName() error {
+	i.log.Infof("Start checking localhostname")
+	hostname, err := i.ops.GetHostname()
+	if err != nil {
+		i.log.Errorf("Failed to get hostname from kernel, err %s", err)
+		return err
+	}
+	if hostname != "localhost" {
+		i.log.Infof("hostname is not localhost, no need to do anything")
+		return nil
+	}
```

Comment:
Can it be empty? In some weird corner case

Reply:
no,we are using os.Hostname function which eventually goes through all the possible OS permutations to get the hostname: name, proc, etc', and in case there was no error during any of the system calls, the hostname should be populated